flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,185 @@
+                            3: 	'Connection Refused, Server unavailable - The Network Connection has been made but the MQTT service is unavailable',
+                            4: 	'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+                            5: 	'Connection Refused, not authorized - The Client is not authorized to connect' }
+
+        MQTT_proto_map = { 'MQTTv31' : mqtt.MQTTv311,
+                           'MQTTv311': 	mqtt.MQTTv31 }
+
+        logger.trace('MQTT notify config={}',str(config))
+
+        client = mqtt.Client(protocol=MQTT_proto_map.get(config.get('broker_protocol',mqtt.MQTTv311)), transport=config.get('broker_transport', 'tcp') )
+
+        client.enable_logger(logger=logger)
+
+        client.on_log = on_log_cb
+        client.on_publish = on_publish_cb
+        client.on_disconnect = on_disconnect_cb
+
+        #Handle SSL/TLS communication w/out certificate authentication
+        if not config.get('certificates',{}).get('client_cert',False) and config.get('enable_encrypted_communication',False):
+            client.tls_set(ca_certs=certs.get('broker_ca_cert'),
+                          certfile=None,
+                          keyfile=None,
+                          cert_reqs=ssl.CERT_NONE)
+
+            client.tls_insecure_set(True)
+
+            logger.verbose('Basic SSL/TLS encrypted communications enabled')
+            logger.verbose('TLS insecure cert mode enabled. Broker cert will not be validated')
+
+
+        #Handle SSL/TLS communication with certificate authentication
+        if config.get('certificates',False):
+            certs = config.get('certificates',{})
+            logger.debug('TLS certificate config: {}',str(certs))
+
+            tls_version_map = {'tlsv1.2': ssl.PROTOCOL_TLSv1_2, 'tlsv1.1': ssl.PROTOCOL_TLSv1_1, 'tlsv1': ssl.PROTOCOL_TLSv1, '': None}
+            tls_version = tls_version_map.get(certs.get('tls_version'),ssl.PROTOCOL_TLSv1_2)
+            logger.verbose('TLS version is {}',str(tls_version))
+
+            cert_required = ssl.CERT_REQUIRED if certs.get('validate_broker_cert', True) else ssl.CERT_NONE
+
+            client.tls_set(ca_certs=certs.get('broker_ca_cert'),
+                          certfile=certs.get('client_cert'),
+                          keyfile=certs.get('client_key'),
+                          cert_reqs=cert_required,
+                          tls_version=tls_version)
+
+            if not certs.get('validate_broker_cert'):
+                client.tls_insecure_set(True)
+                logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+            else:
+                logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+        #Handle user/pass authentication
+        if config.get('username',False) or config.get('password',False):
```

Comment:
1. The default is already `None`, so using `False` is redundant.
2. Note the the current schema enables passing either password, username, both of them or none of them. So using an `or` might not be sufficient.
You could use the [`dependencies`](https://json-schema.org/understanding-json-schema/reference/object.html#dependencies) attribute of jsonschema to help with this.

Reply:
@liiight  
#1 - Addressed with commit 4d0db68
#2 - Added a schema dependency as suggested and simplified if statement logic via commit 826ff3d . The only disallowed combo is a password without specifying a username

Intended allowed combinations are:
- No user/No pass
- Yes User/No pass
- Yes User/Yes pass